### PR TITLE
TypeScript.py: Add TypeScript language definition

### PR DESCRIPTION
**Pull request by margobra8**
This adds a definition for the TypeScript language
so that coala can work with it.

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: coalib/bearlib/languages/definitions/TypeScript.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+
+@Language
+class TypeScript:
+    aliases = 'ts'
+    extensions = '.ts', '.tsx'
+    comment_delimiter = '//'
+    multiline_comment_delimiters = {'/*': '*/'}
+    string_delimiters = {'"': '"', "'": "'"}
+    multiline_string_delimiters = {}
```
Comment (adtac) [inline coalib/bearlib/languages/definitions/TypeScript.py]: https://stackoverflow.com/questions/35225399/multiline-string-with-typescript-angular2 tells me typescript has multiline string support with backticks?

**Comment by corona10** [comment]
@adtac 
FYI, This is the task of GCI, I will approve the task of this student when this PR is merged :)

File: coalib/bearlib/languages/definitions/TypeScript.py (hunk 1 above)
Comment (nkprince007) [reply coalib/bearlib/languages/definitions/TypeScript.py]: Yes, it does.

**Comment by adtac** [comment]
@corona10 of course, I'm aware. Feel free to do so :)

File: coalib/bearlib/languages/definitions/TypeScript.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    indent_types = {'{': '}'}
+    encapsulators = {'(': ')', '[': ']'}
+    keywords = [
+                    'break', 'case', 'catch', 'class', 'const',
```
Comment (nkprince007) [inline coalib/bearlib/languages/definitions/TypeScript.py]: `async` and `await` should also be in the list.

**Comment by jayvdb** [comment]
unack https://github.com/coala/coala/pull/4920/commits/4942b5926a6d1fb1c703203b4d72a21288aa8d9f

File: coalib/bearlib/languages/definitions/TypeScript.py (hunk 1 above)
Comment (margobra8) [reply coalib/bearlib/languages/definitions/TypeScript.py]: Ok will add them later.

**Comment by adtac** [comment]
@margobra8 we require changes to be made within the same commit. Ideally, you would have made your changes and then done a `git add path/to/TypeScript.py` and then done a `git commit --amend`. However, it seems like you did a `git commit` without the `--amend`.

You can fix this with a git squash to squash these two commits into one commit. You can find some documentation on how to do it over here: http://api.coala.io/en/latest/Developers/Git_Basics.html#squashing-your-commits - if you ever get stuck, please feel free to ask for help via chat (on gitter or zulip).

**Comment by margobra8** [comment]
@adtac, oh I never knew about `--amend` despite using for a few years now! Thanks for the tip! Will amend that right now.

**Comment by margobra8** [comment]
@corona10 I think this is all good now.

**Comment by adtac** [comment]
ack https://github.com/coala/coala/pull/4920/commits/a4b2e94b7832105b93bd9f52ca04235a6ef6b9f8

**Comment by adtac** [comment]
@rultor merge

**Comment by corona10** [comment]
@margobra8 

Please submit the GCI task. 👍

**Comment by margobra8** [comment]
@corona10 done! ✅ thanks for the help!

**Pull request merged**